docs/jumpstart/week-10.md

```diff
@@ -0,0 +1,221 @@
+
+### Install Heroku
+The way our application is structured, our frontend and backend are separated. They are not dependent on one another and are developed in different environments and servers. Therefore, we have to deploy our frontend and backend separately. Thankfully, Heroku provides great documentation as to how to deploy our React frontend application and our Java backend. However, before you deploy any application on Heroku we need to install Heroku on our machine. First, we need to create an account on Heroku. The link to sign up is https://signup.heroku.com/. Once you have signed up for a Heroku account, you have to install it on your machine. Because there are different installation methods depending on the machine that you are using, the download instructions for every operating system can be found here https://devcenter.heroku.com/articles/heroku-cli. What you are installing is Heroku's command line application which will allow you to open a terminal and run heroku commands on the terminal.
+
+### Deploy Frontend
+To deploy our frontend application, we need to navigate to the directory that contains our Frontend application. Once we have navigated to our directory we have to initialize a git repository if a git repository has not been initialized yet. To initialize a git repository we run the command:
+ ```shell
+ git init
+ ```
+ After running our git init command, we have to add our project's files to the git repository. To add the files we type the command:
+  ```shell
+  git add .
+  ```
+Once all the files are added we can call
+```shell
+git commit -m 'commit message'
+```
+
+Once we have committed our code, we now have to connect our downloaded Heroku application with our Heroku account. We run the command:
+```shell 
+heroku login
+```
+follow the instructions to enter our Heroku credentials. Once we have logged in, we now have to create a new Heroku project. 
+
+To create a project we type the command:
+```shell 
+heroku create (PROJECT_NAME) --buildpack mars/create-react-app
+```
+By doing this, Heroku is creating a new deployment server for us and giving us a custom URL for us to use. The buidlpack part is our way of asserting
+that our project is a React project. Usually, Heroku automatically detects that it is a React application but sometimes it does not. We add the buildpack just to avoid any errors
+in case. Now comes the part where we connect our application with our new Heroku created project. 
+```shell
+git remote add heroku
+```
+This will add Heroku to our git repository which is how we will push our code to Heroku. Finally, to push our code we run the command: 
+```shell
+git push heroku master
+```
+This will push our current committed code that is in our <strong> master </strong> branch on git. If we want to deploy a branch that is not 
+the master's branch we can run the command:
+```shell
+git push heroku <branch_name>:master
```

Comment:
Maybe add a note in a comment block below this that the command pushes the branch and pretends the branch's name is master